wifi/sender/main.c

```diff
@@ -42,6 +42,12 @@ sw@simonwunderlich.de
 
 #include <opendroneid.h>
 
+/* convert a timespec to a double.
+ * if tv_sec > 2, then inevitable loss of precision in tv_nsec
+ * so best to NEVER use TSTONS()
+ * WARNING replacing 1e9 with NS_IN_SEC causes loss of precision */
+#define TSTONS(ts) ((double)((ts)->tv_sec + ((ts)->tv_nsec / 1e9)))
```

Comment:
I initially got confused by this since I thought 1e9 would be an integer (and thus the division would be zero almost always). However, apparently it expands to a double and everything is then good. Again I learned something :-)

Reply:
I was actually a bit confused by the name TSTONS, since it actually gives the value in seconds, not nanoseconds. Plus, the comments and warnings are not very confidence-inspiring, either. But since this is what's used in the gpsd project, I hope it will do its duty for opendroneid as well ;)